app/src/test/kotlin/compiler/parser/grammar/ParserGrammarTest.kt

```diff
@@ -0,0 +1,9 @@
+package compiler.parser.grammar
+
+import kotlin.test.Test
+
+class ParserGrammarTest {
+    @Test fun `test grammar regexes are correct syntactically`() {
+        ParserGrammar.getGrammar()
+    }
```

Comment:
Maybe a test where we run the grammar through the parser contructor could be useful so we know it doesn't break. Could be in this file or in the parser.integration_tests package.